### PR TITLE
Make API name validation to allow ASCII printable chars

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- Make API name validation to allow ASCII printable chars

**Related issue(s)**

See also #652

### CODE REVIEW

**Comment by dbadura** [comment]
Please update the docs:
https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-input-validation.md

File: components/director/pkg/graphql/api_validation.go [hunk 1]
```diff
@@ -2,13 +2,14 @@ package graphql
...
 	validation "github.com/go-ozzo/ozzo-validation"
+	"github.com/go-ozzo/ozzo-validation/is"
 	"github.com/kyma-incubator/compass/components/director/pkg/inputvalidation"
 	"github.com/pkg/errors"
 )
 
 func (i APIDefinitionInput) Validate() error {
 	return validation.ValidateStruct(&i,
-		validation.Field(&i.Name, validation.Required, inputvalidation.Name),
+		validation.Field(&i.Name, validation.Required, is.PrintableASCII, validation.Length(1, 100)),
```
Comment (dbadura) [inline components/director/pkg/graphql/api_validation.go]: Why 100?
Why not 128?

File: docs/compass/03-input-validation.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ This document contains validation rules for all input types.
 ## Validation rules explanation
 
 - `name` - Up to 36 characters long. Cannot start with digit. The characters allowed in names are: digits (`0`-`9`), lower case letters (`a`-`z`),`-`, and `.`. Based on Kubernetes resource name format.
```
Comment (majakurcius) [inline docs/compass/03-input-validation.md]:
```suggestion
- `name` - Up to 36 characters long. Cannot start with a digit. The characters allowed in names are: digits (`0`-`9`), lower case letters (`a`-`z`),`-`, and `.`. Based on Kubernetes resource name format.
```

**Pull request merged**